system/lib/libc/musl/arch/emscripten/atomic_arch.h

```diff
@@ -36,9 +36,9 @@ static inline void a_or_l(volatile void *p, long v)
 #define a_cas_p a_cas_p
 static inline void *a_cas_p(volatile void *p, void *t, void *s)
 {
-	void* expected = t;
-	__c11_atomic_compare_exchange_strong((_Atomic uintptr_t*)p, &expected, s, __ATOMIC_SEQ_CST, __ATOMIC_SEQ_CST);
-	return expected;
+	uintptr_t expected = (uintptr_t)t;
+	__c11_atomic_compare_exchange_strong((_Atomic uintptr_t*)p, &expected, (uintptr_t)s, __ATOMIC_SEQ_CST, __ATOMIC_SEQ_CST);
+	return (void*)expected;
```

Comment:
Are these atomic changes related?

Reply:
This is needed so that the header is compatible with C++..  this is musl internal header and it normally nevers goes through the C++ compiler.

I could split `struct_info_internal.json` into two.. one for C headers and one for C++ headers.. but it seemed better/simpler to fix this header to avoid the C++ error.